client/src/components/Doctor/DoctorAppointments.tsx

```diff
@@ -1,122 +1,153 @@
-import React, { FunctionComponent, useContext } from "react";
-import { useNavigate } from "react-router-dom";
-import { AuthContext } from "../../providers/AuthProvider";
+import axios from 'axios';
+import React, { FunctionComponent, useContext, useEffect, useState } from 'react';
+
+import { AuthContext } from '../../providers/AuthProvider';
+interface AppointmentDetails {
+  name: string;
+  patient_email: string;
+  appointment_id: string;
+  patient_name: string;
+  created_at: string;
+  appointment_time: string;
+  appointment_status: string;
+}
 
 const DoctorAppointments: FunctionComponent<{}> = () => {
-  const appointments = [
-    {
-      name: "Shruthi",
-      email: "shruthi@gmail.com",
-      date: "03/26/2023",
-      time: "9.00 AM",
-      status: "Pending",
-      id: "1",
-    },
-    {
-      name: "Hasini",
-      email: "hasini@gmail.com",
-      date: "03/27/2023",
-      time: "10.00 AM",
-      status: "Confirmed",
-      id: "1",
-    },
-  ];
+  const { user, role, logout } = useContext(AuthContext);
+  const [appointmentData, setAppointmentData] = useState<AppointmentDetails[]>([]);
+
+  useEffect(() => {
+    getAppointment();
+  }, []);
+  const getAppointment = async () => {
+    let resp = await axios.get('/getAppointments/' + user?.email);
```

Comment:
please use authfetch implementation which adds the jwt header directly on the front end.